When user has a single domain that has email, auto-redirect to management screen

fredrikekelund opened the pull request:
#### Proposed Changes

When users have a single domain and they have email for that domain, we should automatically take them to `/email/:domain/manage/:site_slug` as opposed to `/email/:site_slug` when they click Upgrades > Email in the sidebar.

#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Have a site with a single domain and an email solution for that domain
2. Click Upgrades > Emails in the sidebar
3. Ensure that you are taken directly to `/email/:domain/manage/:site_slug`

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

fredrikekelund committed:
When user has a single domain that has email, auto-redirect to management screen

olaseni commented on client/my-sites/email/email-management/email-home.tsx:
```diff
@@ -173,14 +178,15 @@ const EmailHome = ( props: EmailManagementHomeProps ): ReactElement => {
-	) {
-		page( emailManagementTitanSetUpMailbox( selectedSite?.slug, selectedSite?.domain ) );
+	if ( isSingleDomainThatHasEmail ) {
+		if (
+			( domainsWithEmail[ 0 ].titanMailSubscription?.maximumMailboxCount ?? 0 ) > 0 &&
+			domainsWithEmail[ 0 ].titanMailSubscription?.numberOfMailboxes === 0
+		) {
+			page( emailManagementTitanSetUpMailbox( selectedSite?.slug, selectedSite?.domain ) );
+		} else {
+			page( emailManagement( selectedSite?.slug, selectedSite.domain ) );
+		}
```
```suggestion
		if (
			( domainsWithEmail[ 0 ].titanMailSubscription?.maximumMailboxCount ?? 0 ) > 0 &&
			domainsWithEmail[ 0 ].titanMailSubscription?.numberOfMailboxes === 0
		) {
			page( emailManagementTitanSetUpMailbox( selectedSite?.slug, selectedSite?.domain ) );
			return;
		}
		
		page( emailManagement( selectedSite?.slug, selectedSite.domain ) );
		
```

What do you think about simplifying like this?

fredrikekelund replied:
Readability-wise, I don't really mind either way. But I went ahead and changed it!

olaseni commented on client/my-sites/email/email-management/email-home.tsx:
```diff
@@ -141,13 +148,16 @@ const EmailHome = ( props: EmailManagementHomeProps ): ReactElement => {
-				<EmailPlan selectedSite={ selectedSite } domain={ selectedDomain } source={ source } />
+				<EmailPlan
+					domain={ selectedDomain }
+					hideHeaderCake={ isSingleDomainThatHasEmail }
```
Could you leave a note on why we are hiding the `HeaderCake`. This should make it easy to follow the changes in the future.

fredrikekelund replied:
I don't mind adding comments, but I don't think this is an especially complex scenario and the comment might be a bit verbose compared to what I seem to usually find in Calypso.

If you think this comment is good, then let's go with it 🙂

olaseni reviewed (approved):
This works as described!

Since we redirected to the page, should we fix the back button though? Otherwise it feels odd that I should be taken "back" to a page I never visited in the first place.

This is non-blocking and can probably be referenced in a new PR if needed, but I'm leaving a video for illustration.

https://user-images.githubusercontent.com/277661/175921621-7623322c-866e-4d6d-b61e-e318dcae68aa.mov

fredrikekelund committed:
Fix issue with non-custom domains

When a site was not using the custom domain as the primary domain, the redirection logic would not kick in. This fixes that.

fredrikekelund committed:
Address review comments

fredrikekelund commented:
@olaseni I've updated the PR to make sure that it works regardless of the site's primary domain settings. If you could test one more time, that'd be great!

fredrikekelund committed:
Fix bad redirect logic
